### PR TITLE
Add more info to efolder failure logs

### PR DESCRIPTION
**Pull request by lowellrex**
We have been getting many different `Caseflow::Error::EfolderAccessForbidden` errors over the last few days (example: https://sentry.ds.va.gov/department-of-veterans-affairs/caseflow/issues/1458/). These could come in [a few different flavors](https://github.com/department-of-veterans-affairs/caseflow-efolder/blob/aa10043cf7033edf08dde9a9fa09c9b47b68d967/app/controllers/api/v1/application_controller.rb#L30). This PR adds some more logging when we see that error so we can better determine what the root cause is.

### CODE REVIEW

File: app/services/external_api/efolder_service.rb [hunk 1]
```diff
@@ -64,11 +64,11 @@ def self.check_for_error(response_body:, code:, vbms_id:, user_id:)
...
       end
     when 403
-      fail Caseflow::Error::EfolderAccessForbidden, "403"
+      fail Caseflow::Error::EfolderAccessForbidden, "403: #{response_body}"
     when 400
-      fail Caseflow::Error::ClientRequestError, "400"
+      fail Caseflow::Error::ClientRequestError, "400: #{response_body}"
     when 500
-      fail Caseflow::Error::DocumentRetrievalError, "502"
+      fail Caseflow::Error::DocumentRetrievalError, "502: #{response_body}"
```
Comment (aroltsch) [inline app/services/external_api/efolder_service.rb]: I think this change will break Reader error response: https://github.com/department-of-veterans-affairs/caseflow/blob/master/app/controllers/reader/documents_controller.rb#L65. In particularly this code
```
status: e.message
```
We have to fix Reader controller to parse error properly.

File: app/controllers/reader/documents_controller.rb [hunk 2]
```diff
@@ -62,7 +62,11 @@ def documents
   end
 
   def respond_to_doc_retrieval_error(e)
-    render json: { "errors": ["status": e.message, "title": e.to_s, "detail": e.message] }, status: e.message
+    if e.class.method_defined? :serialize_response
+      render e.serialize_response
+    else
+      render json: { "errors": ["status": e.message, "title": e.to_s, "detail": e.message] }, status: e.message
+    end
```
Comment (lowellrex) [inline app/controllers/reader/documents_controller.rb]: Changed to allow errors to serialize themselves if they know how. Doing this makes it so this function does not need to know anything about the error object itself (while in the else condition this function does need to know about the internal properties/methods of the error object).

File: lib/caseflow/error.rb [hunk 3]
```diff
@@ -1,8 +1,26 @@
...
+        @message = args[:message]
+      else
+        @code = args
+        @message = args
+      end
+    end
+
+    def serialize_response
+      { json: { "errors": ["status": @code, "title": @message, "detail": @message] }, status: @code }
+    end
```
Comment (lowellrex) [inline lib/caseflow/error.rb]: Mirror how we would respond in `Reader::DocumentsController. respond_to_doc_retrieval_error()`.

File: lib/caseflow/error.rb [hunk 4]
```diff
@@ -1,8 +1,26 @@
...
+    attr_accessor :code, :message
+
+    def initialize(args)
+      if args.is_a?(Hash)
+        @code = args[:code]
+        @message = args[:message]
+      else
+        @code = args
+        @message = args
+      end
```
Comment (lowellrex) [inline lib/caseflow/error.rb]: Allow the existing errors to continue to behave exactly as they did before this change.

**Pull request merged**